build_daemon/lib/daemon.dart

```diff
@@ -89,6 +91,24 @@ class Daemon {
     }));
   }
 
+  BuildMode _buildModeFor(Set<String> options) {
+    for (var option in options) {
+      if (option.startsWith('--$buildModeFlag')) {
+        var modeValue = option.split('=').last;
```

Comment:
There are other valid ways of providing this arg, it is probably worth just using the arg parser here imo? But we should at least support the `--build-mode <mode>` style.

Reply:
There are no other valid ways since the client actually embeds this argument and it does it exactly as parsed.

Reply:
That seems like a pretty strong assumption to make still?